ansible/roles/vm_set/files/mux_simulator.py

```diff
@@ -28,6 +28,50 @@
 LOWER_TOR = 'lower_tor'
 NIC = 'nic'
 
+# Global variable for storing mux cable flapping
+flap_counter = {}
+
+def get_flap_counter(vm_set, port_index):
+    """
+    @summary: Get the flap counter for a certain mbr
+    @param vm_set: A str, vm_set name
+    @param port_index: An integer, the port_index
+    @return: An dict {'mbr-vms17-8-0': 10}
+    """
+    mbr_name = "mbr-{}-{}".format(vm_set, port_index)
+    return {mbr_name: flap_counter.get(mbr_name, 0)}
+
+
+def increase_flap_counter(vm_set, port_index):
+    """
+    @summary: Increase the flap counter for a certain mbr by 1
+    @param vm_set: A str, vm_set name
+    @param port_index: An integer, the port_index
+    @return: None
+    """
+    mbr_name = "mbr-{}-{}".format(vm_set, port_index)
+    flap_counter[mbr_name] = flap_counter.get(mbr_name, 0) + 1
+
+
+def clear_flap_counter(vm_set, port_index):
+    """
+    @summary: Clear the flap counter for a certain mbr
+    @param vm_set: A str, vm_set name
+    @param port_index: An integer, the port_index
+    @return: None
+    """
+    mbr_name = "mbr-{}-{}".format(vm_set, port_index)
+    flap_counter[mbr_name] = 0
+
+
+def fill_flap_counter():
```

Comment:
A better name might be `init_flap_counter`? 'Fill' seems a little ambiguous

Reply:
Good suggestion. Updated. Thanks